sql-plugin/src/test/scala/com/nvidia/spark/rapids/HostAllocSuite.scala

```diff
@@ -695,14 +695,31 @@ class HostAllocSuite extends AnyFunSuite with BeforeAndAfterEach with
     class AllocOnAnotherThreadWithWarmup(override val thread: TaskThread,
         override val size: Long) extends AllocOnAnotherThread(thread, size) {
 
+      def wasteSomeTime(): Unit = {
+        val startTime = System.currentTimeMillis
+        val duration = 100 // 100ms
+
+        val endTime = startTime + duration
+        var result = 0.0
+
+        while (System.currentTimeMillis < endTime) {
+          result += Math.sqrt(Math.random)
```

Comment:
On JDK9+ we would call `Thread.onSpinWait` here. Maybe use `Thread.yield` here until we are upgraded in terms of JDK

Reply:
good idea, I have changed to using yield. thx @gerashegalov !